zuul-core/src/main/java/com/netflix/zuul/netty/server/http2/Http2ContentLengthEnforcingHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.zuul.netty.server.http2;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.codec.http2.DefaultHttp2ResetFrame;
+import io.netty.handler.codec.http2.Http2Error;
+import java.util.List;
+
+/**
+ * This class is only suitable for use on HTTP/2 child channels.
+ */
+public final class Http2ContentLengthEnforcingHandler extends ChannelInboundHandlerAdapter {
+    private static final long UNSET_CONTENT_LENGTH = -1;
+
+    private long expectedContentLength = UNSET_CONTENT_LENGTH;
+
+    private long seenContentLength;
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (msg instanceof HttpRequest) {
+            HttpRequest req = (HttpRequest) msg;
+            List<String> lengthHeaders = req.headers().getAll(HttpHeaderNames.CONTENT_LENGTH);
+            if (lengthHeaders.size() > 1) {
+                ctx.writeAndFlush(new DefaultHttp2ResetFrame(Http2Error.PROTOCOL_ERROR));
+                return;
+            } else if (lengthHeaders.size() == 1) {
+                expectedContentLength = Long.parseLong(lengthHeaders.get(0));
+                if (expectedContentLength < 0) {
+                    // TODO(carl-mastrangelo): this is not right, but meh.  Fix this to return a proper 400.
+                    ctx.writeAndFlush(new DefaultHttp2ResetFrame(Http2Error.PROTOCOL_ERROR));
+                    return;
+                }
+            }
+            if (expectedContentLength != UNSET_CONTENT_LENGTH && HttpUtil.isTransferEncodingChunked(req)) {
```

Comment:
Consider extracting usages of `expectedContentLength != UNSET_CONTENT_LENGTH` and naming appropriately to convey the intent of that check.

Reply:
Done.